hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestAsyncTableScanMetrics.java

```diff
@@ -70,7 +72,7 @@ public class TestAsyncTableScanMetrics {
 
   @FunctionalInterface
   private interface ScanWithMetrics {
-    Pair<List<Result>, ScanMetrics> scan(Scan scan) throws Exception;
+    Pair<List<Result>, Pair<ScanMetrics, List<ScanMetrics>>> scan(Scan scan) throws Exception;
```

Comment:
A Pair of Pairs? It's time to make a static POJO class or use record type (I guess the former because want to backport to branch-2 lineages).